doc/OTA_app_system_document.md

```diff
@@ -0,0 +1,123 @@
+
+### Update Scenario
+
+[2022-07-22] OTA app proivdes updating a new application (not driver component) at runtime currently. Since flashing applications should follow `MPU alignment rule`, it is only possible to update an application which has the size smaller or equal than the size of OTA app (8198 bytes). If an application which don't follow `MPU alignment rule` is flahsed, the loaded application will be erased. Furtuermore, if the number of application loaded on the target board reaches to the maximum number of application that the target board can run, OTA app doesn't execute update.
+
+### Module Dependency
+[2022-07-22] The following image describes `ota app` module dependency implemented on Tock. We assume that `ota_uart.send.py` act as web server which send data to IoT device. From the external tool to `console.rs`, binary data move through `①`. When receiving the specified size of data (517), console callback function `②` is triggered. Then `ota_app` parses the receiving data and do actions according to a command which is positioned at index 0 of the data. After completing actions corresponding commands, `ota_app` send `ota_uart_send.py` a coressponding response for next sequence. An application (.tbf) binary is written to flash memory via `③`. When writing the binary is done, `ota_send_uart.py` delivers crc32 value which it sends to `ota_app`, and `ota_app` also calculates the data that it received. Then the app request `process_load_utilities.rs` to calculate the written binary data into flash memory, and return the resulting value to `ota-app` via `④ and ⑤`. `ota_app` checks whether or not the three crc32 values are same. If there is incorrect crc32 consistency, `ota_app` erases the loaded data. When the update procedure passes the crc32 consistency state, `ota_app` requests loading the loaded application through ⑥. If the loaded app doesn't follow `MPU alignment rule`, `ota_app` erases the loaded data and don't load the entry point of the loaded application into `PROCESS global array` at main.rs.
+  
+
+![ota app module dependency]](ota_app_module_dependency.png)
+
+### Key points
+ 
+[2022-07-22] Dynamically changing start address of flash memory and sram.
+When we update an application at runtime, we don't have to interfere flash and sram memory region which is occupied by the kernel and other apps. If we commit such memory access violation, the system is going to be crushed. To prevent this issue, there are three key variables that save the dynamically changing unused flash memory and sram memory start address at "process_load_utilities.rs". 
+
+At `find_dynamic_start_address_of_writable_flash_advanced`, it parses an address immediately next to the last application in flash memory and an index used to save the entry point of the loaded application into PROCESS global array. Then it saves the address and index to `dynamic_flash_start_addr` and `index` at grant region separately.
+
+
+```c
```

Comment:
These should be rust code blocks, not c

Also, can you wrap the documentation lines to 80 charecters

Reply:
Thank you for your feedback! I applied it to my doc